src/Translation_unit.ml

```diff
@@ -230,6 +230,13 @@ let check_all_locations fmt cmts_t =
           "Warning: Some locations have not been considered\n%!" ;
         List.iter ~f:print (List.sort l ~compare:Location.compare)
 
+let check_margin (conf : Conf.t) ~filename ~fmted =
+  if conf.margin_check then
+    List.iteri (String.split_lines fmted) ~f:(fun i line ->
+        if String.length line > conf.margin then
+          Format.fprintf Format.err_formatter
+            "Warning: %s:%i exceeds the margin\n%!" filename i)
```

Comment:
This does not print warnings the same way as `Compat.print_warning`. That would require using ocaml's internal warning system, though. Maybe we need our own formatting function for warning/error messages ?